library/src/main/java/com/alamkanak/weekview/WeekView.java

```diff
@@ -465,6 +490,74 @@ public boolean onScale(ScaleGestureDetector detector) {
         });
     }
 
+    private void resetHomeDate()
+    {
+        Calendar newHomeDate = today();
+
+        if(mMinDate != null && newHomeDate.before(mMinDate)) {
+            newHomeDate = (Calendar) mMinDate.clone();
+        }
+        if(mMaxDate != null && newHomeDate.after(mMaxDate)) {
+            newHomeDate = (Calendar) mMaxDate.clone();
+        }
+
+        if (mMaxDate != null) {
+            Calendar date = (Calendar) mMaxDate.clone();
+            date.add(Calendar.DATE, 1-mNumberOfVisibleDays);
+            while(date.before(mMinDate)) {
+                date.add(Calendar.DATE, 1);
+            }
+
+            if(newHomeDate.after(date)) {
+                newHomeDate = date;
+            }
+        }
+
+        mHomeDate = newHomeDate;
+    }
+
+    private float getXOriginForDate(Calendar date) {
+        int dateDifference = (int) (date.getTimeInMillis() - mHomeDate.getTimeInMillis()) / (1000 * 60 * 60 * 24);
```

Comment:
I think it should be  
change
```int dateDifference = (int) (date.getTimeInMillis() - mHomeDate.getTimeInMillis()) / (1000 * 60 * 60 * 24);```
to 
```
int dateDifference = (int)((date.getTimeInMillis() - mHomeDate.getTimeInMillis()) / (1000 * 60 * 60 * 24));
```
because: date.getTimeInMillis() - mHomeDate.getTimeInMillis() is a long value, it can be greater than Interger.MAX.